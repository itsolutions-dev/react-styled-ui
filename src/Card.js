import styled, { css } from './styledComponent';
import {
  getSizeOrDefault,
} from './utils/';

const Card = styled({
  component: 'div',
  style: css`
    display: flex;
    flex-direction: column;
    overflow: hidden;
    width: ${props => getSizeOrDefault(props.width, '300px')};
    border-radius: 4px;
    box-shadow: inset 0 0 0px 0px #dee1e3, 0 0 14px #dee1e3;
  `,
});

export default Card;
