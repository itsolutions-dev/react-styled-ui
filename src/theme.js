export default {
  radius: 8,
  spacing: {
    margin: 8,
    padding: 8,
    iconSize: 24,
    borderSize: 1,
    shadowSize: 1,
  },
  font: {
    size: 16,
    weight: 200,
    lineHeight: 18,
    family: 'Roboto, sans-serif',
  },
  h1: {
    size: 32,
    weight: 900,
    lineHeight: 64,
  },
  h2: {
    size: 24,
    weight: 700,
    lineHeight: 32,
  },
  h3: {
    size: 18,
    weight: 500,
    lineHeight: 24,
  },
  selected: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  hover: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  transitions: {
    in: undefined,
    out: undefined,
  },
  palette: {
    default: 'green',
    defaultText: 'white',
    defaultDisabled: undefined,
    defaultHover: undefined,
    defaultSelected: undefined,
    primary: 'red',
    primaryText: 'white',
    primaryDisabled: undefined,
    primaryHover: undefined,
    primarySelected: undefined,
    secondary: 'blue',
    secondaryText: 'white',
    secondaryDisabled: undefined,
    secondaryHover: undefined,
    secondarySelected: undefined,
    disabled: 'lightgray',
    shadow: 'black',
  },
};
