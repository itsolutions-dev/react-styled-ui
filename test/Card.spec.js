import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import theme from '../src/theme';
import Card from '../src/Card';

describe('Card', () => {
  test('should render a Card', () => {
    const component = ReactTestRenderer.create(<Card theme={theme} />);
    expect(component).toBeDefined();
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should have a div tag', () => {
    const component = ReactTestRenderer.create(<Card theme={theme} />).toJSON();
    expect(component.type).toEqual('div');
  });

  test('should have a default width', () => {
    const component = ReactTestRenderer.create(
      <Card
        theme={theme}
      />,
    );
    expect(component).toHaveStyleRule('width', '300px');
  });

  test('should handle string width', () => {
    const component = ReactTestRenderer.create(
      <Card
        theme={theme}
        width="20em"
      />,
    );
    expect(component).toHaveStyleRule('width', '20em');
  });

  test('should handle numeric width', () => {
    const component = ReactTestRenderer.create(
      <Card
        theme={theme}
        width={30}
      />,
    );
    expect(component).toHaveStyleRule('width', '30px');
  });
});
