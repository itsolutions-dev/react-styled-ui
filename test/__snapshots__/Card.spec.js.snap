// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render a Card 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
  width: 300px;
  border-radius: 4px;
  box-shadow: inset 0 0 0px 0px #dee1e3,0 0 14px #dee1e3;
}

<div
  className="c0"
/>
`;
